pkg/distributor/push.go

```diff
@@ -190,9 +214,23 @@ func toHTTPStatus(ctx context.Context, pushErr error, limits *validation.Overrid
 	return http.StatusInternalServerError
 }
 
-func addHeaders(w http.ResponseWriter, err error) {
+func addHeaders(w http.ResponseWriter, err error, r *http.Request, responseCode int, retryCfg RetryConfig) {
 	var doNotLogError middleware.DoNotLogError
 	if errors.As(err, &doNotLogError) {
 		w.Header().Set(server.DoNotLogErrorHeaderKey, "true")
 	}
+
+	if responseCode == http.StatusTooManyRequests || responseCode/100 == 5 {
+		var retrySeconds string
+		var retryAttempt string
+		if retryCfg.Enabled {
+			retryAttempt = r.Header.Get("Retry-Attempt")
+			retrySeconds = calculateRetryAfter(retryAttempt, retryCfg)
```

Comment:
```suggestion
		if retryCfg.Enabled {
			retryAttempt := r.Header.Get("Retry-Attempt")
			retrySeconds := calculateRetryAfter(retryAttempt, retryCfg)
```

Reply:
I'd also suggest to passing only base and maxAttempts to `calculateRetryAfter`: `calculateRetryAfter(retryAttempt.BaseSeconds, retryCfg.MaxAllowedAttempts)`

Reply:
I don't have strong opinion on this. but I was thinking about that, it would means we need to leave the logics of cap retry-attemps and handle it errors of converstion outside of functions, I am not sure that it is better for unitest.

Reply:
i see what you mean, expanded retryCfg to pass base and maxbackoffexponent